pandas/io/parsers.py

```diff
@@ -997,14 +997,14 @@ def _validate_usecols_arg(usecols):
 
     if usecols is not None:
         if callable(usecols):
-            return usecols
+            return usecols, None
         usecols_dtype = lib.infer_dtype(usecols)
         if usecols_dtype not in ('empty', 'integer',
                                  'string', 'unicode'):
             raise ValueError(msg)
 
-        return set(usecols)
-    return usecols
+        return set(usecols), usecols_dtype
```

Comment:
why does this still have a ``set`` conversion here? is this for a different case? if so, then should happen in your validation below (to make it more consistent), IOW, this *always* returns  tuple of (list of usecols, dtype of usecols). Please make the doc-string indicate that as well.